kyo-prelude/shared/src/main/scala/kyo2/Result.scala

```diff
@@ -76,39 +88,40 @@ object Result:
         def isPanic: Boolean =
             self.isInstanceOf[Panic]
 
-        inline def fold[B](inline ifError: E => B)(inline ifPanic: Throwable => B)(inline ifSuccess: A => B): B =
-            (self: @unchecked) match
-                case self: Error[E] =>
-                    ifError(self.error)
-                case self: Panic =>
-                    ifPanic(self.exception)
+        inline def fold[B](inline ifFailure: Failure[E] => B)(inline ifSuccess: A => B): B =
```

Comment:
I've just pushed an additional commit simplifying `fold` to take two functions. Other methods like `recover` use a similar approach.